### PR TITLE
added --exit parameter to exit whenever an error occurred

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
You can now use --exit to shut down the miner whenever an error occurred.
Note, that this is not recommended, and only useful for people with additional software to restart the miner
https://github.com/ethereum-mining/ethminer/issues/97

### CODE REVIEW

**Comment by DeadManWalkingTO** [comment]
Love it!

File: ethminer/MinerAux.h [hunk 1]
```diff
@@ -627,9 +633,11 @@ class MinerCLI
 			<< "        1: eth-proxy compatible: dwarfpool, f2pool, nanopool (required for hashrate reporting to work with nanopool)" << endl
 			<< "        2: EthereumStratum/1.0.0: nicehash" << endl
 			<< "    -RH, --report-hashrate Report current hashrate to pool (please only enable on pools supporting this)" << endl
+			<< "    -HWMON Displays gpu temp and fan percent." << endl
```
Comment (XhmikosR) [inline ethminer/MinerAux.h]: @MariusVanDerWijden: I think this is wrong here?

Comment (MariusVanDerWijden) [reply ethminer/MinerAux.h]: This was the old version, I didn't change it... It's a merge error. I'll resolve it

**Comment by sandrodz** [comment]
This doesn't seem to work in case of driver crash, when hash drops to zero:

https://github.com/ethereum-mining/ethminer/issues/891

**Comment by agbarbosa** [comment]
Is there a possibility to create a "system event" when the hashrate is detected as zero during the ethminer execution?  Users could later create their own treatment for the events (i.e.: TaskScheduler for Windows based on the Event ID).

**Comment by agbarbosa** [comment]
Guys, --exit is great, but it would be better if it could cause a System Event for Windows machines so it could be captured by Event Log.

**Pull request merged**